compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/expressions/impl/IrCallImpl.kt

```diff
@@ -54,7 +54,7 @@ class IrCallImpl(
             type: IrType,
             symbol: IrSimpleFunctionSymbol,
             typeArgumentsCount: Int = symbol.descriptor.typeParametersCount,
-            valueArgumentsCount: Int = symbol.descriptor.valueParameters.size,
+            valueArgumentsCount: Int = symbol.descriptor.valueParameters.size + symbol.descriptor.contextReceiverParameters.size,
```

Comment:
Did I miss anything, or you don't actually store the information about context receivers in corresponding function declarations? We'll need it for proper signature mapping on JVM.